#10

ThilinaTCH opened the pull request:
#10 Add cTree (drawer) search support

CSchmitz81 commented on src/ctree-app/ctree-app.html:
```diff
@@ -185,6 +190,11 @@
       }
+      paper-toggle-button.green {
+        --paper-toggle-button-checked-bar-color:  var(--paper-green-500);
+        --paper-toggle-button-checked-button-color:  var(--paper-green-500);
```
Please try to keep consistent code formatting (there are 2 spaces before `var` instead of 1)

CSchmitz81 commented on src/ctree-app/ctree-app.html:
```diff
@@ -212,21 +222,25 @@
-          <paper-input label="Search" no-label-float>
+          <paper-input label="Search" value="{{searchText}}" no-label-float>
             <iron-icon icon="search" slot="prefix"></iron-icon>
+            <paper-icon-button slot="suffix" on-tap="clearSearch" icon="clear" alt="clear" title="clear"></paper-icon-button>
```
This should only be visible if text has been entered, and it should disappear when the field is cleared.

CSchmitz81 commented on src/ctree-app/ctree-app.html:
```diff
@@ -212,21 +222,25 @@
-              </template>
-            </iron-list>
+                <template is="dom-if" if="[[searchText]]">
+                  <paper-toggle-button class="green" checked="{{item.data.saved}}" title="save"></paper-toggle-button>
```
I'm not seeing this yet.  It should appear for all items when `searchText` is not empty.

CSchmitz81 commented on src/ctree-app/ctree-app.html:
```diff
@@ -373,8 +394,8 @@
 
-      _rootChanged(path) {
-        this._updateNavList();
+      clearSearch(){
```
This should probably be marked as private by starting with `_` because I don't think we'll need to procedurally clear the search.

CSchmitz81 commented on src/ctree-app/ctree-app.html:
```diff
@@ -397,7 +418,7 @@
       }
 
-      _updateNavList() {
+      getCTreeList() {
```
This should probably be marked as private by starting with `_`.

CSchmitz81 commented on src/ctree-app/ctree-app.html:
```diff
@@ -411,9 +432,19 @@
+                newNavList.push({name: name, data: pageData});
+            }
+          }else{
+            if(pageData.saved){
```
Please follow the code style conventions of the rest of the code and put spaces between keywords like `if` and `else`, and operators like `(` and `{`.  For example `if (this.searchText) {` and `} else {`.

CSchmitz81 commented on src/ctree-app/ctree-app.html:
```diff
@@ -411,9 +432,19 @@
-          newNavList.push({name: name, data: pageData});
+
+          if(this.searchText){
+            if(pageData.label.toLowerCase().indexOf(this.searchText.toLowerCase()) !== -1){
```
We may want to get the lower case search text before we start the for loop so we don't need to regenerate it for each tree.

CSchmitz81 commented on polymer.json:
```diff
@@ -14,6 +14,6 @@
   ],
   "extraDependencies": [
-    "bower_components/webcomponentsjs/webcomponents-lite.min.js"
+    "bower_components/webcomponentsjs/*.js"
```
I think we don't want to do this because it could make the download cost much larger.  Is there a specific reason for this?  It looks like maybe we should have just changed this to `webcomponents-lite.js` (not sure where the `.min` comes from)

ThilinaTCH replied:
Yes, actually in new plymer-cli the build failed here

CSchmitz81 commented on bower.json:
```diff
@@ -54,7 +54,8 @@
     "polymer": "polymer/polymer#^2.0.0",
-    "webcomponentsjs": "webcomponents/webcomponentsjs#^1.0.0"
+    "webcomponentsjs": "webcomponents/webcomponentsjs#^1.0.0",
+    "paper-toggle-button": "^2.0.0"
```
I think we've sorted these alphabetically, so please add this at the correct location.

CSchmitz81 commented:
OK, I reviewed the functionality and the toggle button didn't show up.  Also the X button never disappeared.  I added a css style so hidden would work for the X button, but the toggle never showed up.  Even if it had, that's not the style we wanted anyway (we use icons everywhere), so I updated it to use a bookmark icon.  I didn't change the color, but feel free to add that and fix the tooltip I tried to add (didn't work, but if you can't fix it quickly don't worry about it).  What I left for you is to tie up the logic to lookup the ctree when the bookmark button is clicked and toggle the saved state.  That should finish up this issue.

ThilinaTCH commented:
Hi Chris

Everything working fine for me, I guess there is an  issue with the polymer
versions we are using. Can you try by updating polymer.-cli version


On Mon, Dec 25, 2017 at 5:32 AM Chris Schmitz <notifications@github.com>
wrote:

> OK, I reviewed the functionality and the toggle button didn't show up.
> Also the X button never disappeared. I added a css style so hidden would
> work for the X button, but the toggle never showed up. Even if it had,
> that's not the style we wanted anyway (we use icons everywhere), so I
> updated it to use a bookmark icon. I didn't change the color, but feel free
> to add that and fix the tooltip I tried to add (didn't work, but if you
> can't fix it quickly don't worry about it). What I left for you is to tie
> up the logic to lookup the ctree when the bookmark button is clicked and
> toggle the saved state. That should finish up this issue.
>
> —
> You are receiving this because you authored the thread.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/F4IF/ctree-demo/pull/47#issuecomment-353808440>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADdTvo1sR-RqG8sPJ_jy4V6cJjd4rppwks5tDuYxgaJpZM4Q54KK>
> .
>
-- 
Thanks and Regards,
Thilina

CSchmitz81 commented:
OK, maybe it not working with your paper-toggle-button was an issue with my CLI.  That said, we don't use paper-toggle-button because it doesn't fit with the style, which is why I updated it to paper-icon-button (to match our other bookmark icons).  That's why this required implementing on-tap.  That said, after looking again at what you did, it was cleaner and appeared to actually fulfill the requirements of the issue, so I added a new component to work like the paper-toggle-button (ctree-toggle-button) and completed this implementation.  Please use (and update if needed) ctree-toggle-button instead of paper-toggle-button for any future changes unless you've confirmed using paper-toggle-button with me first.

Now that this is working I'll approve and merge.

ThilinaTCH commented:
Hi Chris,

Ok will use ctree-toggle-button if needed
Great to hear that my PR is approved
Can you please put a review on my Fiverr order too ☺
I will update you on next task Im going to work soon

Thanks
Thilina

On Fri, Dec 29, 2017 at 12:53 PM, Chris Schmitz <notifications@github.com>
wrote:

> OK, maybe it not working with your paper-toggle-button was an issue with
> my CLI. That said, we don't use paper-toggle-button because it doesn't fit
> with the style, which is why I updated it to paper-icon-button (to match
> our other bookmark icons). That's why this required implementing on-tap.
> That said, after looking again at what you did, it was cleaner and appeared
> to actually fulfill the requirements of the issue, so I added a new
> component to work like the paper-toggle-button (ctree-toggle-button) and
> completed this implementation. Please use (and update if needed)
> ctree-toggle-button instead of paper-toggle-button for any future changes
> unless you've confirmed using paper-toggle-button with me first.
>
> Now that this is working I'll approve and merge.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/F4IF/ctree-demo/pull/47#issuecomment-354397815>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADdTvoVLn8tX9fnzBkCdUGrI2n06u6Thks5tFHBfgaJpZM4Q54KK>
> .
>



-- 
Thanks and Regards,
Thilina
